exchanges/okx/okx_wrapper.go

```diff
@@ -520,22 +520,23 @@ func (ok *Okx) UpdateAccountInfo(ctx context.Context, assetType asset.Item) (acc
 	if !ok.SupportsAsset(assetType) {
 		return info, fmt.Errorf("%w: %v", asset.ErrNotSupported, assetType)
 	}
-	balances, err := ok.GetBalance(ctx, "")
+	accountBalances, err := ok.GetNonZeroBalances(ctx, "")
 	if err != nil {
 		return info, err
 	}
-	currencyBalance := make([]account.Balance, len(balances))
-	for i := range balances {
-		free := balances[i].AvailBal
-		locked := balances[i].FrozenBalance
-		currencyBalance[i] = account.Balance{
-			Currency: currency.NewCode(balances[i].Currency),
-			Total:    balances[i].Balance,
-			Hold:     locked,
-			Free:     free,
+	currencyBalances := []account.Balance{}
+	for _, a := range accountBalances {
+		// Not clear from API docs why we'd get multiple records back in this array; Asked Okx for clarification.
+		for _, d := range a.Details {
```

Comment:
Just seems a standard for their API and I don't think this comment is necessary

Reply:
Happy to remove the comment.

Don't think we want to ignore the possibility of handling the array, though. Agreed?